skorch/tests/test_helper.py

```diff
@@ -450,12 +451,13 @@ def test_grid_search_with_slds_works(
         gs = GridSearchCV(
             net, params, refit=False, cv=3, scoring='accuracy', error_score='raise'
         )
-        gs.fit(slds, y)  # does not raise
+        gs.fit(slds, to_numpy(y))  # does not raise
```

Comment:
Not supporting a mixture of NumPy arrays and CPU pytorch arrays, feels like a regression from scikit-learn's side.

Reply:
For me, the question is: is it an intended change or was it introduced accidentally?

If it's unintended, one way to resolve this would be here:

https://github.com/scikit-learn/scikit-learn/blob/1922303a79aa776768e2ee89bbda5b6eb4dd5d8b/sklearn/utils/_array_api.py#L177

There would need to be another check

1. if one of the devices is an instance of `torch.device` 
2. and for this device, `device.type == "cpu"` 
3. and the other device is `"cpu"`

it should be allowed.

Reply:
Looks like there is already a bug report and a PR to fix it in scikit-learn:

https://github.com/scikit-learn/scikit-learn/issues/29107
https://github.com/scikit-learn/scikit-learn/pull/30454